mlrun/api/api/endpoints/marketplace.py

```diff
@@ -67,6 +67,9 @@ async def create_source(
     response_model=List[IndexedMarketplaceSource],
 )
 async def list_sources(
+    item_name: Optional[str] = Query(None),
+    tag: Optional[str] = Query(None),
+    version: Optional[str] = Query(None),
```

Comment:
Since `tag` and `version` are only supported when `item_name` is provided, then please verify that they are `None` if it wasn't provided, or explode.